[RFC] vim-patch:8.2.{0953,0955,0956,1678}

vigoux opened the pull request:
Hopefully I did it right, this is following [this request](https://github.com/neovim/neovim/issues/12949).

#### vim-patch:8.2.0953: spell checking doesn't work for CamelCased words

Problem:    Spell checking doesn't work for CamelCased words.
Solution:   Add the "camel" value in the new option 'spelloptions'.
            (closes vim/vim#1235)
https://github.com/vim/vim/commit/362b44bd4aa87a2aef0f8fd5a28d68dd09a7d909


#### vim-patch:8.2.0955: build fails

Problem:    Build fails.
Solution:   Add missing struct change.
https://github.com/vim/vim/commit/215f49c4d720367be29f56168f363a0ee9eaf86b


#### vim-patch:8.2.0956: spell test fails

Problem:    Spell test fails.
Solution:   Add missing change the spell checking.
https://github.com/vim/vim/commit/e0ebeda4d8219a8955001b43ef7a9268452ef7f5

#### vim-patch:8.2.1678: crash when using ":set" after ":ownsyntax"

Problem:    Crash when using ":set" after ":ownsyntax".
Solution:   Make sure 'spelloptions' is not NULL. (closes vim/vim#6950)
vim/vim@d1f76af

vigoux committed:
vim-patch:8.2.0953: spell checking doesn't work for CamelCased words

Problem:    Spell checking doesn't work for CamelCased words.
Solution:   Add the "camel" value in the new option 'spelloptions'.
            (closes vim/vim#1235)
https://github.com/vim/vim/commit/362b44bd4aa87a2aef0f8fd5a28d68dd09a7d909

vigoux committed:
vim-patch:8.2.0956: spell test fails

Problem:    Spell test fails.
Solution:   Add missing change the spell checking.
https://github.com/vim/vim/commit/e0ebeda4d8219a8955001b43ef7a9268452ef7f5

vigoux committed:
vim-patch:8.2.0955: build fails

Problem:    Build fails.
Solution:   Add missing struct change.
https://github.com/vim/vim/commit/215f49c4d720367be29f56168f363a0ee9eaf86b

vigoux commented:
Appveyor fail seems unrelated.

vigoux commented:
@janlazo may I have your opinion on this ? This is my very first port...

janlazo commented on runtime/doc/options.txt:
```diff
@@ -5672,6 +5672,16 @@ A jump table for the options with a short description can be found at |Q_op|.
+						*'spelloptions'* *'spo'*
+'spelloptions' 'spo'	string	(default "")
+			local to buffer
+			{not available when compiled without the |+syntax|
```
`+syntax` feature is always enabled.

janlazo commented on src/nvim/spell.c:
```diff
@@ -362,6 +362,8 @@ size_t spell_check(
   size_t wrongcaplen = 0;
   int lpi;
   bool count_word = docount;
+  int use_camel_case = *wp->w_s->b_p_spo != NUL;
```
bool

janlazo commented on src/nvim/spell.c:
```diff
@@ -362,6 +362,8 @@ size_t spell_check(
   int lpi;
   bool count_word = docount;
+  int use_camel_case = *wp->w_s->b_p_spo != NUL;
+  int camel_case = 0;
```
bool

janlazo reviewed (changes requested):
LGTM aside from the minor issues.

janlazo commented:
Please port patch 8.2.1678 as well.

vigoux committed:
vim-patch:8.2.1678: crash when using ":set" after ":ownsyntax"

Problem:    Crash when using ":set" after ":ownsyntax".
Solution:   Make sure 'spelloptions' is not NULL. (closes vim/vim#6950)
https://github.com/vim/vim/commit/d1f76afaf963be706697279ab0570ffcb8a1f2fc

vigoux commented:
All comments adressed, @janlazo may I ask you to review again ?

janlazo commented on src/nvim/spell.c:
```diff
@@ -362,6 +362,8 @@ size_t spell_check(
   int lpi;
   bool count_word = docount;
+  bool use_camel_case = *wp->w_s->b_p_spo != NUL;
+  bool camel_case = 0;
```
false

vigoux replied:
Lol I am so lazy sometimes...

bluz71 commented:
@vigoux,

I just tested the latest Neovim `master` branch, the new `set spelloptions=camel` options works great.

Thank you very much.

Best regards.
